### PR TITLE
kubevirt: add VMWare Import provider to CreateVMWizard

### PR DESCRIPTION
**Pull request by suomiy**
- shows status, vcenter selection (or adding a new vcenter secret)
- VMWare VM select which loads and prefills the dialog with VM data
- fix memoization issue when usign context
- small UI fixes for errors and modals
- fix wizard closing and css
- add interop for v2v vmware
- add import source to disk modal only when importing
- add correctVCenterSecretLabels patches
- add v2vVmware and v2vvmware deployment statuses
- fix some v2v UI bugs (status, data loading)
- add PVC type
- toShallowJS takes firehose results into account
- added support for just models when comparing references
- refactored ID references - it is enough to update changedProps just in create-vm-wizard (providers just need to update IDReferences in the store)

### CODE REVIEW

**Comment by suomiy** [comment]
screenshots 

a


b

c

d

@matthewcarleton @mareklibra

**Comment by mareklibra** [comment]
Thank you for porting this from the kubevirt-web-ui-components here and adapting to new design! The screenshots looks fantastic.

What is the reason for moving the import out of the `Create VM Wizard`, please?

Considering the size of this PR, I try to run it before reading through the code.
It's failing for me. I have not dig into details yet, so this is recent observation only:
- the `v2vvmware` deployment passes well, but I keep getting HTTP 404 for ` http://localhost:9000/api/kubernetes/apis/apps/v1/deployments/v2v-vmware` . When looking at the deployment resource, the self link is: `/apis/extensions/v1beta1/namespaces/default/deployments/v2v-vmware`.
- Once the VMware connection details are entered, the `secret` object is created but not fetched by UI (HTTP 404 again for `http://localhost:9000/api/kubernetes/api/v1/secrets/administrator-vcenter.v2v.bos-trck8`, but self link is `/api/v1/namespaces/default/secrets/administrator-vcenter.v2v.bos-trck8`)

---

Note for future readers: redux-store extension mechanism is WIP, so to make this working, hack it by adding `kubevirt: kubevirtReducer` to `combineReducers()` in `frontend/public/redux.ts`.

**Comment by suomiy** [comment]
> Thank you for porting this from the kubevirt-web-ui-components here and adapting to new design! The screenshots looks fantastic.

thanks

> 
> What is the reason for moving the import out of the `Create VM Wizard`, please?

It has its own dialog according to the new design: http://openshift.github.io/openshift-origin-design/web-console/knikubevirt/import-vm/import-vm

> 
> Considering the size of this PR, I try to run it before reading through the code.
> It's failing for me. I have not dig into details yet, so this is recent observation only:
> 
> * the `v2vvmware` deployment passes well, but I keep getting HTTP 404 for ` http://localhost:9000/api/kubernetes/apis/apps/v1/deployments/v2v-vmware` . When looking at the deployment resource, the self link is: `/apis/extensions/v1beta1/namespaces/default/deployments/v2v-vmware`.
> * Once the VMware connection details are entered, the `secret` object is created but not fetched by UI (HTTP 404 again for `http://localhost:9000/api/kubernetes/api/v1/secrets/administrator-vcenter.v2v.bos-trck8`, but self link is `/api/v1/namespaces/default/secrets/administrator-vcenter.v2v.bos-trck8`)
> 

thanks for finding this, there was a wrong passing of a namespace, although it miracourously worked on my setup. Fixed.

**Comment by matthewcarleton** [comment]
This is looking awesome @suomiy ! I've updated [this doc](https://docs.google.com/document/d/12FAE0Qd6-NtF8DkvX_A2naK7oIDQz2sBqKnGEjc3BE8/edit#heading=h.yfgl9ue2wwha) to track my feedback

File: frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/form/form-field-row.tsx [hunk 1]
```diff
@@ -8,50 +8,88 @@ import { FormRow } from '../../form/form-row';
...
+type FieldFormInnerRowProps = {
+  field: any;
+  fieldType: FormFieldType;
+  children?: React.ReactNode;
+  loadingResources?: { [k: string]: any };
+  validation?: ValidationObject;
+};
+
+export const FormFieldInnerMemoRow = React.memo(
+  FormFieldInnerRow,
```
Comment (mareklibra) [inline frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/form/form-field-row.tsx]: I am wondering what kind of issue this "double-memoization" is trying to solve. If the `FormFieldInnerRow` is not a pure function, wouldn't this cause unexpected skips in rendering?

**Pull request merged**